src/index.js

```diff
@@ -3,16 +3,27 @@ export default middleware
 
 
 export { runSaga } from './internal/runSaga'
-export { CANCEL } from './internal/proc'
 export { END, eventChannel, channel } from './internal/channel'
 export { buffers } from './internal/buffers'
 export { takeEvery, takeLatest } from './internal/sagaHelpers'
-export { delay } from './internal/utils'
 
+import { CANCEL } from './internal/proc'
 import * as effects from './effects'
 import * as utils from './utils'
 
+export function delay(ms, val=true) {
+  let timeoutId;
+  const promise = new Promise((resolve) => {
+    timeoutId = setTimeout(resolve, ms, val);
```

Comment:
That's correct. Although I'm not sure about current browser compatibility requirements for `redux-saga`, so maybe I should revert it.